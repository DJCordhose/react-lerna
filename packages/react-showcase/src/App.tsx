import React from "react";
import logo from "./logo.svg";
import { Counter } from "./features/counter/Counter";
import "./App.css";

import { useTranslation } from "react-i18next";

function App() {
  const { t, i18n } = useTranslation();

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <Counter />
        <p>
          {i18n.language !== "de" && (
            <button onClick={() => i18n.changeLanguage("de")}>{t("de")}</button>
          )}
          {i18n.language !== "en" && (
            <button onClick={() => i18n.changeLanguage("en")}>{t("en")}</button>
          )}
        </p>
        <p>{t("users", {
          date: new Date(),
          count: 1_000_000,
          revenue: 1_000_000_000.95
        })}</p>
      </header>
    </div>
  );
}

export default App;
