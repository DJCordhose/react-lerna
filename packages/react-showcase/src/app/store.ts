import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});

// https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
