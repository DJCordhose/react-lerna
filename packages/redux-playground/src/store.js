import { createSlice, configureStore } from "@reduxjs/toolkit";

const counterSlice = createSlice({
  name: "ounter",
  initialState: {
    value: 0,
  },
  reducers: {
    incremented: (state) => {
      // Redux Toolkit allows us to write "mutating" logic in reducers. It
      // doesn't actually mutate the state because it uses the Immer library,
      // which detects changes to a "draft state" and produces a brand new
      // immutable state based off those changes
      state.value += 1;
    },
    decremented: (state) => {
      state.value -= 1;
    },
  },
});

export const { incremented, decremented } = counterSlice.actions;

export const store = configureStore({
    reducer: counterSlice.reducer
  })
// You can use subscribe() to update the UI in response to state changes.
// Normally you'd use a view binding library (e.g. React Redux) rather than subscribe() directly.
// There may be additional use cases where it's helpful to subscribe as well.

store.subscribe(() => console.log(store.getState()));

export const selectCount = (state) => state.value;
