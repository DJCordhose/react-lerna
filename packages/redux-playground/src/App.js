import logo from './logo.svg';
import './App.css';

import { useSelector, useDispatch } from 'react-redux'

import {DECREMENT, INCREMENT, selectCount} from './store'

function App() {
  const dispatch = useDispatch();
  const count = useSelector(selectCount);
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <p>Counter: {count}</p>
        <button onClick={() => dispatch(DECREMENT)}>Runter</button>
        <button onClick={() => dispatch(INCREMENT)}>Hoch</button>
      </header>
    </div>
  );
}

export default App;
