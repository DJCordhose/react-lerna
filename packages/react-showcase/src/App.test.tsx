import React, { Suspense } from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "./app/store";
import App from "./App";

// https://react.i18next.com/misc/testing
jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

test("smoke: renders something", () => {
  const { getByText } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(getByText(/users/i)).toBeInTheDocument();
});

// https://jestjs.io/docs/en/snapshot-testing
test("snapshot: renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
    
  expect(tree).toMatchSnapshot();
});
