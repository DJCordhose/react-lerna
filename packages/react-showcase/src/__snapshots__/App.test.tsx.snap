// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: renders correctly 1`] = `
<div
  className="p-6 max-w-sm mx-auto space-x-5"
  id="cnt-app"
>
  <div
    className="flex justify-end"
  >
    <button
      aria-label="Change Language"
      className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
      data-testid="language:switch:de"
      onClick={[Function]}
    >
      de
    </button>
    <button
      aria-label="Change Language"
      className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
      data-testid="language:switch:en"
      onClick={[Function]}
    >
      en
    </button>
  </div>
  <div
    className=""
  >
    <div
      className="flex justify-between  border-gray-900 border-2 m-2 overflow-hidden"
    >
      <button
        aria-label="Increment value"
        className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
        data-testid="count:increment"
        onClick={[Function]}
      >
        +
      </button>
      <span
        aria-label="Value"
        className="px-5 py-2 border-gray-900 border-2  m-2"
        data-testid="count:value"
        role="presentation"
      >
        0
      </span>
      <button
        aria-label="Decrement value"
        className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
        data-testid="count:decrement"
        onClick={[Function]}
      >
        -
      </button>
    </div>
    <div
      className="flex justify-between border-gray-900 border-2 m-2"
    >
      <input
        aria-label="Set increment amount"
        className="px-2 py-2 m-2 w-20"
        onChange={[Function]}
        type="number"
        value={2}
      />
      <button
        aria-label="add"
        className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
        data-testid="count:incrementByAmount"
        onClick={[Function]}
      >
        add
      </button>
      <button
        aria-label="add"
        className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-20 mx-2 my-2 "
        data-testid="count:incrementByAmountAsync"
        onClick={[Function]}
      >
        add
      </button>
    </div>
    <div
      className="flex justify-center border-gray-900 border-2 m-2"
    >
      <button
        aria-label="load"
        className="
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-100 mx-2 my-2 "
        data-testid="count:load"
        onClick={[Function]}
      >
        load
      </button>
    </div>
  </div>
  <p>
    users
  </p>
</div>
`;
